src/components/filter.rs

```diff
@@ -0,0 +1,75 @@
+use sycamore::prelude::*;
+
+// table structure
+#[component]
+pub fn Filter<'a, G: Html>(
+    cx: Scope<'a>,
+) -> View<G> {
+    view! { cx,
+        div (class="flex flex-col shadow-md rounded-md ") {
+            // filter title
+            div (class="bg-primary p-4 font-bold rounded-t-md") { 
+                p(class="text-primary-content text-lg") { "Filter" }
+            }
+        
+            // Term
+            div(class="flex rounded-b-md") {
+                FilterSection(title="Term".to_string())
+                
+            }
+            div(class="flex rounded-b-md") {
+                FilterSection(title="Time".to_string())
+                
+            }
+        }       
+    }        
+}
+
+// filter section
+#[derive(Prop)]
+pub struct FilterScetionProps {
+    title: String,
+    // TODO: allow multiple filter selections 
+    // names: Vec<String>,
+}
+
+#[component]
+pub fn FilterSection<'a, G: Html>(
+    cx: Scope<'a>,
+    FilterScetionProps { title }: FilterScetionProps,
+) -> View<G> {
+    view! {cx,
+        div(class="relative p-4 bg-base-100 rounded-b-md w-full") {
+            h3 (class="text-lg font-semibold") { (title) }
+            div (class="flex justify-center") {
+                div (class="flex flex-col w-1/2") {
```

Comment:
It is possible to define all of these filters right now and connect them to a state. 

You might find this example helpful: https://github.com/sycamore-rs/sycamore/blob/abd556cbc02047042dad2ebd04405e455a9b11b2/examples/todomvc/src/main.rs

There is also a video here: https://www.youtube.com/watch?v=jpKILQ3-Jq0

Please define all the filters that will be in the final UI